api/src/test/java/com/epam/pipeline/manager/pipeline/RunScheduleManagerTest.java

```diff
@@ -103,214 +117,238 @@ public void setUp() {
         testPipeline.setOwner(TEST_NAME);
         testPipeline.setRepositorySsh(TEST_PIPELINE_REPO_SSH);
-        pipelineDao.createPipeline(testPipeline);
+        testPipeline.setId(RUN_ID);
 
         createPipelineRun(RUN_ID, testPipeline.getId());
+
         runConfiguration = createRunConfiguration();
+        runConfiguration.setId(2L);
 
         PipelineUserVO userVO = new PipelineUserVO();
         userVO.setUserName(USER_OWNER);
-        userManager.createUser(userVO);
 
         testRunScheduleVO = getRunScheduleVO(RunScheduledAction.PAUSE, CRON_EXPRESSION1);
         testRunScheduleVO2 = getRunScheduleVO(RunScheduledAction.RESUME, CRON_EXPRESSION2);
         testRunScheduleVO3 = getRunScheduleVO(RunScheduledAction.RUN, CRON_EXPRESSION2);
 
+        MockitoAnnotations.initMocks(this);
+
+        mockRunConfigurationScheduleProvider =
+                Mockito.spy(new RunConfigurationScheduleProvider(mockMessageHelper, mockRunConfigurationManager));
+
+        doReturn(mockPipelineRunScheduleProvider)
+                .when(mockScheduleProviderManager).getProvider(ScheduleType.PIPELINE_RUN);
+        doReturn(mockRunConfigurationScheduleProvider)
+                .when(mockScheduleProviderManager).getProvider(ScheduleType.RUN_CONFIGURATION);
+        doReturn(mockPipelineUser).when(mockAuthManager).getCurrentUser();
     }
 
+
     @Test
-    @WithMockUser(username = USER_OWNER)
-    public void testCreateRunSchedules() {
-        final List<RunSchedule> runSchedules = runScheduleManager.createSchedules(RUN_ID, ScheduleType.PIPELINE_RUN,
-                    Arrays.asList(testRunScheduleVO, testRunScheduleVO2));
-        runSchedules.forEach(this::loadAndAssertSchedule);
+    public void testCreatePipelineRunSchedules() {
+        runScheduleManager.createSchedules(RUN_ID, ScheduleType.PIPELINE_RUN,
+                Arrays.asList(testRunScheduleVO, testRunScheduleVO2));
 
-        List<RunSchedule> configSchedules = runScheduleManager.createSchedules(runConfiguration.getId(),
-                ScheduleType.RUN_CONFIGURATION, Collections.singletonList(testRunScheduleVO3));
-        configSchedules.forEach(this::loadAndAssertSchedule);
+        verify(mockRunScheduler, times(2)).scheduleRunSchedule(any());
+        verify(mockRunScheduleDao, times(1)).createRunSchedules(anyList());
+        verify(mockScheduleProviderManager, times(3)).getProvider(ScheduleType.PIPELINE_RUN);
+    }
+
+
+    @Test
+    public void testCreateRunConfigurationSchedules() {
+        doReturn(runConfiguration).when(mockRunConfigurationManager).load(anyLong());
 
-        Mockito.verify(runScheduler, Mockito.times(3)).scheduleRunSchedule(any());
+        runScheduleManager.createSchedules(runConfiguration.getId(),
+                ScheduleType.RUN_CONFIGURATION, Collections.singletonList(testRunScheduleVO3));
 
+        verify(mockRunScheduler, times(1)).scheduleRunSchedule(any());
+        verify(mockRunScheduleDao, times(1)).createRunSchedules(anyList());
```

Comment:
It's a nice idea to skip extra `times(1)` argument because `verify(something, times(1))` is equivalent to `verify(something)`. You can check the verify method's implementation to be sure about that.